src/storage-client/src/controller.rs

```diff
@@ -844,6 +856,10 @@ pub enum StorageError {
     },
     /// Dataflow was not able to process a request
     DataflowError(DataflowError),
+    InalterableSource {
+        id: GlobalId,
+        halt: bool,
```

Comment:
who should do what with this `halt`? I can look at the code, but I'm also lazy and rustdoc would help... 😅

Reply:
Removed the interior halt from the error and added an explicit halt in `alter_collection`.